commit 5f9814f5ebd7af90ede6c0219097b64e161b60c5 (refs/changes/63/1463/4)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-25T01:38:23+00:00 (2 years, 7 months ago)
    
    CNS-195 error handling mixups in dns-server.js
    Reviewed by: Richard Kiene <richard.kiene@joyent.com>

diff --git a/lib/dns-server.js b/lib/dns-server.js
index bc9a23b..03510b1 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -528,19 +528,22 @@ DNSServer.prototype.getAllZones = function (r, cb) {
 	var zs = Object.keys(this.config.forward_zones);
 
 	r.keys('zone:*.arpa', function (err, keys) {
-		if (err || keys === null)
-			return;
-
-		for (var i = 0; i < keys.length; ++i) {
-			var k = keys[i].split(':')[1];
-			zs.push(k);
+		if (!err && keys !== null) {
+			for (var i = 0; i < keys.length; ++i) {
+				var k = keys[i].split(':')[1];
+				zs.push(k);
+			}
 		}
 
 		vasync.forEachParallel({
 			func: getZoneSerial,
 			inputs: zs
 		}, function (err2, res) {
-			cb(err2, res.successes);
+			if (err2) {
+				cb(err2);
+				return;
+			}
+			cb(null, res.successes);
 		});
 
 		function getZoneSerial(z, scb) {
@@ -548,8 +551,8 @@ DNSServer.prototype.getAllZones = function (r, cb) {
 			r.get(zk, function (err3, serStr) {
 				var res = {};
 				res.zone = z;
-				if (err) {
-					scb(err);
+				if (err3) {
+					scb(err3);
 					return;
 				}
 				if (serStr === null) {
